comp/forwarder/defaultforwarder/default_forwarder.go

```diff
@@ -519,56 +519,38 @@ func (f *DefaultForwarder) createAdvancedHTTPTransactions(endpoint transaction.E
 
 	for _, payload := range payloads {
 		for domain, dr := range f.domainResolvers {
-			drDomain, destinationType := dr.Resolve(endpoint) // drDomain is the domain with agent version if not local
-
-			if payload.Destination == transaction.LocalOnly {
-				// if it is local payload, we should not send it to the remote endpoint
-				if destinationType == pkgresolver.Local && endpoint == endpoints.SeriesEndpoint {
-					t := transaction.NewHTTPTransaction()
-					t.Domain = drDomain
-					t.Endpoint = endpoint
-					t.Payload = payload
-					t.Priority = priority
-					t.Kind = kind
-					t.StorableOnDisk = storableOnDisk
-					t.Destination = payload.Destination
-					t.Headers.Set("Authorization", fmt.Sprintf("Bearer %s", dr.GetBearerAuthToken()))
-					for key := range extra {
-						t.Headers.Set(key, extra.Get(key))
-					}
-					tlmTxInputCount.Inc(drDomain, endpoint.Name)
-					tlmTxInputBytes.Add(float64(t.GetPayloadSize()), domain, endpoint.Name)
-					transactionsInputCountByEndpoint.Add(endpoint.Name, 1)
-					transactionsInputBytesByEndpoint.Add(endpoint.Name, int64(t.GetPayloadSize()))
-					transactions = append(transactions, t)
+			drDomain, _ := dr.Resolve(endpoint) // drDomain is the domain with agent version if not local
```

Comment:
We can change `Resolve` to only return the Domain now as the destinationType return is not used anywhere.